### PR TITLE
Improve example templates

### PR DESCRIPTION
**Pull request by davidkpiano**
This PR improves the templates to use a new `timeMachine` that fetches the current time and assigns it to `context`:

```ts
const timeMachine = createMachine(
  {
    id: 'time',
    initial: 'idle',
    context: {
      time: undefined
    },
    states: {
      idle: {
        on: {
          fetch: 'pending'
        }
      },
      pending: {
        invoke: {
          src: 'fetchTime',
          onDone: {
            actions: assign({
              time: (_, event) => event.data
            })
          }
        },
        on: {
          cancel: 'idle'
        }
      },
      success: {
        on: {
          refetch: 'pending'
        }
      }
    }
  },
  {
    services: {
      fetchTime: () =>
        new Promise((res) => {
          setTimeout(() => {
            res(new Date().toLocaleTimeString());
          }, 1000);
        })
    }
  }
);
```

The goal of this new template is to provide a statechart example that isn't as contrived as the original toggle example, has more states and transitions, and also shows the usefulness of invoking states.

### CODE REVIEW

File: examples/template-ts/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    initial: 'idle',
+    context: {
+      time: undefined
+    },
+    states: {
+      idle: {
+        on: {
+          fetch: 'pending'
+        }
+      },
```
Comment (KrisCoulson) [inline examples/template-ts/src/index.ts:20]: I really like this example. One thing for clarity might be worth capitalizing states and actions or using some type of differentiation to say that those are user-defined and not special xstate keys. especially when things like fetch already exist in javascript.

File: examples/template-ts/package.json [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
+  "main": "index.html",
   "scripts": {
-    "dev": "vite",
-    "build": "tsc && vite build",
-    "serve": "vite preview"
+    "start": "parcel index.html --open",
+    "build": "parcel build index.html"
+  },
+  "dependencies": {
+    "parcel-bundler": "1.12.5",
```
Comment (Andarist) [inline examples/template-ts/package.json:11]: nit: this usually should be a dev dep

File: examples/template-ts/package.json [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
+    "xstate": "^4.32.1"
   },
   "devDependencies": {
     "typescript": "^4.7.4",
     "vite": "^2.9.13"
```
Comment (Andarist) [inline examples/template-ts/package.json:16]: you have switched to using parcel for this example so vite should be dropped from here

File: packages/core/CHANGELOG.md [hunk 4]
```diff
@@ -503,10 +503,10 @@
 
   model.createMachine({
     // `ctx` was of type `any`
-    entry: ctx => {},
+    entry: (ctx) => {},
```
Comment (Andarist) [inline packages/core/CHANGELOG.md:506]: We are using a different Prettier version than Changesets and this results in a lot of those reformatting changes. I have a PR open that is supposed to fix this: https://github.com/changesets/changesets/pull/905

**Pull request closed without merging**